### DISCUSSION TITLE
Migrate workflow with unsupported features

### DISCUSSION THREAD

**Discussion by darioielardi** [prisma/prisma#6335] [Q&A] [answered]
We have a pretty large codebase for an enterprise product, using Prisma Client for data access, and we're more than happy with it. We're still using knex for migrations though, and we'd like to switch to Migrate now that's production ready. I played with it a bit, and found out we have one big problem: `prisma migrate dev` won't stop generating migrations until it thinks our Prisma schema and the DB schema are in sync. This is generally a good thing, unless the DB schema includes Postgres features which Prisma does not support yet. This has never been a problem with Prisma Client generation, because we don't rely on introspection: we just update the `schema.prisma` manually.

The workflow I expected:
1. `prisma migrate dev --create-only` to generate a migration.
2. Delete unwanted SQL from the generated migration ( for example, `CREATE` statements for DB views declared as simple tables in the `schema.prisma`, or `ALTER` statements for [Postgres generated columns](https://www.postgresql.org/docs/13/ddl-generated-columns.html) ).
3. Run the migration with another command.

The problem is, I need to run `migrate dev` again to run the migration, and that's when it generates ( and run! ) another migration with the SQL I deleted from the previous one.

What's the suggested workflow in situation like this? The only thing I can think of is to run `prisma migrate dev --create-only` to generate migrations and `prisma migrate deploy` to run them. Is this the only way to go?

Thank you for your time and your amazing work.

**Comment by tomhoule** [comment, marked as answer]
Hi @darioielardi! There is currently no way to make these features work nicely with migrate/introspect, to my knowledge. They're just not supported, introspection and migration inference (which is at the base of everything `migrate dev` does) cannot take them into account, so you will get drift and unwanted migrations.

The docs mention a workflow for views that does not work with migrate, it's currently in our plans to change that soon.

In the slightly longer run, we of course want to support computed columns and views — the best way to make it happen is to signal interest in the relevant issues ([1](https://github.com/prisma/prisma/issues/2597), [2](https://github.com/prisma/prisma/issues/678))

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
